src/app/Topology/Toolbar/TopologyFilters.tsx

```diff
@@ -272,7 +272,10 @@ export const TopologyFilter: React.FC<{ isDisabled?: boolean }> = ({ isDisabled,
             value={{
               toString: () => opt,
               compareTo: (other) => {
-                const regex = new RegExp(typeof other === 'string' ? other : other.value, 'i');
+                const regex = new RegExp(
+                  typeof other === 'string' ? other : isKeyValue(other) ? other.value : `${other}`,
```

Comment:
```suggestion
                  typeof other === 'string' ? other : isKeyValue(other) ? keyValueToString(other): `${other}`,
```

Should it be this since value can be the same but different key.